src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -455,7 +473,69 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                     }
 
                     break;
+                case TableExpressionKind.Sort:
```

Comment:
It would be nice to have an optimization where we "reuse" a CTE when we are both filtering and sorting on a search parameter.

Reply:
Yes, I agree, but as we have discussed before this optimization can be implemented in a future iteration.